src/models/Movie.ts

```diff
@@ -0,0 +1,33 @@
+import Ids from '@/models/Ids';
+import FanartApi from '@/services/FanartApi';
+
+export default class Movie {
+    private readonly _title: string;
+    private readonly _year: number;
+    private readonly _ids: Ids;
+    private _thumb: string = '';
+
+    constructor(newTitle: string, newYear: number, newIds: Ids) {
+        this._title = newTitle;
+        this._year = newYear;
+        this._ids = newIds;
+        new FanartApi().getMovieThumb(this._ids.imdb)
+            .then((result) => this._thumb = result);
```

Comment:
This shouldn't be done within a model class - violates single responsibility principle as the class may need to be changed to reflect different method of retrieving thumbnail (e.g. caching etc. in future). 

Surely this property might be accessed before the request completes? In which case only the blank value will be used. 

Will a Movie object always require a thumbnail URL? What if this object is constructed when deserialized e.g. from JSON?

Potential solution: provide a setter for this property and assign it when needed. Provide FanartApi via dependency injection in the class that does this